In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('mobile_events_2020.csv')
df.head()

anonymous_id business_zoned context_app_version  \
0  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
1  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
2  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
3  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
4  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   

                      context_device_id context_os_name  \
0  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
1  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
2  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
3  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
4  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   

                        event_name     received_at      created_at  
0  Out-Of-Business Zone View Shown  9/1/2020 23:59  9/1/2020 23:59  
1               Application Opened  9/1/2020 23:59  9/1/2020 23:59  
2           Ride Done - Successful  9/1/2020 23:59  9/1/2020 23:59  
3               Application Opened  9/1/2020 23:59  9/1/2020 23:37  
4         Ride Done - Unsuccessful  9/1/2020 23:59  9/1/2020 23:37

In [3]:
count_sessions = df.value_counts('event_name')
print(count_sessions)

event_name
Application Opened                     26047
Scooter Selected - Auto                 8735
Scooter Selected - User                 6999
Ride Started - Pre-Ride                 6693
Ride Done - Successful                  5165
                                       ...  
Paypal Dialogue - Cancel Pressed          18
Customer Blocked - Phone Validation       16
Paypal Error Shown                        10
Location Services Enabled                  8
Location Services Dialogue Opened          8
Name: count, Length: 61, dtype: int64


In [5]:
df2 = count_sessions.to_frame()
df2['percent'] = (df2['count']/ df2['count'].sum())*100
df2

count    percent
event_name                                           
Application Opened                   26047  25.431060
Scooter Selected - Auto               8735   8.528441
Scooter Selected - User               6999   6.833493
Ride Started - Pre-Ride               6693   6.534729
Ride Done - Successful                5165   5.042862
...                                    ...        ...
Paypal Dialogue - Cancel Pressed        18   0.017574
Customer Blocked - Phone Validation     16   0.015622
Paypal Error Shown                      10   0.009764
Location Services Enabled                8   0.007811
Location Services Dialogue Opened        8   0.007811

[61 rows x 2 columns]

Question_1: What is the % of sessions with an intention to book a scooter?

Ans:    As per my analysis the %age of sessions with an intention to book a scooter is 25.43%

Question_2: What is the % of sessions with a successful booking?

Ans: The %age of sessions with successful booking is around 15%

Question _3: What determines, according to you, an intention to book?

Ans: According to me, number of time "Application Opened" determines an intention to book.

Question_4: What are factors, according to you, determining a successful booking?

Ans: The factors determining a successful booking are the Scooter selected (Auto and User) and I have not considered canceled or unsuccessful rides as the question asks only for booking and not completed rides or successful rides 